Add `await` to `bot.add_cog()` call

RMednis opened the pull request:
Since Red 3.5 the `bot.add_cog()` function is now asynchronous and needs to be awaited for the cog to be successfully registered.

This change literally just adds an `await` for it.

RMednis committed:
`bot.add_cog()` is now asynchronous

Since Red 3.5 the `bot.add_cog()` function is now asynchronous and needs to be awaited for the cog to be successfully registered.

This change literally just adds an `await` for it.

Jackenmen reviewed (changes requested):
Could you update the min_bot_version to 3.5.0 in info.json as well?

RMednis committed:
Changed the `min_bot_version` to require at least Red 3.5.0

As Jackenmen mentioned, since we're changing the `__init__.py` to match Red 3.5.0, we should also bump the minimum required version!

Jackenmen reviewed (approved):
Looks good, thanks!
